tests/bulkwritecommand/bulkwritecommand-updateOne_error-003.phpt

```diff
@@ -0,0 +1,57 @@
+--TEST--
+MongoDB\Driver\BulkWriteCommand::updateOne() with invalid options
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/basic.inc';
+
+$bulk = new MongoDB\Driver\BulkWriteCommand;
+
+// @TODO: Invalid key 'y': update only works with $ operators and pipelines
+echo throws(function() use ($bulk) {
+    $bulk->updateOne(NS, ['x' => 1], ['y' => 1], ['multi' => true]);
+}, 'MongoDB\Driver\Exception\InvalidArgumentException'), "\n\n";
+
+echo throws(function() use ($bulk) {
+    $bulk->updateOne(NS, ['x' => 1], ['y' => 1], ['collation' => 1]);
+}, 'MongoDB\Driver\Exception\InvalidArgumentException'), "\n\n";
+
+echo throws(function() use ($bulk) {
+    $bulk->updateOne(NS, ['x' => 1], ['$set' => ['y' => 1]], ['collation' => 1]);
+}, 'MongoDB\Driver\Exception\InvalidArgumentException'), "\n\n";
+
+echo throws(function() use ($bulk) {
+    $bulk->updateOne(NS, ['x' => 1], ['$set' => ['y' => 1]], ['arrayFilters' => 1]);
+}, 'MongoDB\Driver\Exception\InvalidArgumentException'), "\n\n";
+
+// @TODO: Expected "arrayFilters" option to yield array but got non-sequential keys
+echo throws(function() use ($bulk) {
+    $bulk->updateOne(NS, ['x' => 1], ['$set' => ['y' => 1]], ['arrayFilters' => ['foo' => 'bar']]);
+}, 'MongoDB\Driver\Exception\InvalidArgumentException'), "\n\n";
```

Comment:
The error isn't the same as with collection bulk write: 
- Collection BulkWrite: `"arrayFilters" option has invalid keys for a BSON array`
- Client BulkWriteCommand: `Expected "arrayFilters" option to yield array but got non-sequential keys`

https://github.com/mongodb/mongo-php-driver/blob/d3a079fb4d6f9173a0c81c0d0250d1304e0292a3/tests/bulk/bulkwrite-update_error-003.phpt#L26-L28

Reply:
This is probably due to newer (read, better) logic in client bulk write. We could change the collection bulk write to use the same exception message, but I'm not sure it's worth it.